File: TestTasks/src/task1/src/observer_node.cpp
#include "ros/ros.h"
#include <geometry_msgs/PoseStamped.h>

using namespace ros;

void poseCallback(const geometry_msgs::PoseStamped::ConstPtr& msg)
{
  ROS_INFO("Runner pose: x=%f, y=%f, z=%f", msg->pose.position.x, msg->pose.position.y, msg->pose.position.z);
}

int main(int argc, char **argv)
{
  init(argc, argv, "observer_node");
  NodeHandle nh;
  Subscriber pose_sub = nh.subscribe("pose", 1000, poseCallback);

  Rate loop_rate(10);

  while (ok())
  {
    spinOnce();
    loop_rate.sleep();
  }

  return 0;
}

File: TestTasks/src/task1/src/observer_node.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import PoseStamped


def pose_callback(msg):
    rospy.loginfo("Runner pose: x=%f, y=%f, z=%f", msg.pose.position.x, msg.pose.position.y, msg.pose.position.z)


def main():
    rospy.init_node('observer_node', anonymous=True)
    rate = rospy.Rate(10)
    pose_sub = rospy.Subscriber('pose', PoseStamped, pose_callback)

    while not rospy.is_shutdown():
        rate.sleep()


if __name__ == '__main__':
    main()
